crates/solidity/inputs/language/src/definition.rs

```diff
@@ -3671,44 +3671,44 @@ codegen_language_macros::compile!(Language(
                         Token(
                             name = DecimalLiteral,
                             definitions = [
-                                // An integer (without a dot or a fraction) is enabled in all versions:
-                                TokenDefinition(
-                                    scanner = TrailingContext(
-                                        scanner = Sequence([
-                                            Fragment(DecimalDigits),
-                                            Optional(Fragment(DecimalExponent))
-                                        ]),
-                                        not_followed_by = Fragment(IdentifierStart)
-                                    )
-                                ),
-                                // An integer and a dot (without a fraction) is disabled in "0.5.0"
+                                // An integer, optionally with a dot that may not be followed by a fraction, is disabled in "0.5.0":
```

Comment:
The idea behind splitting each `Token` into multiple `TokenDefinition`s is that each can have their own versioning (for post-processing), but they all should still be mutually exclusive/non-ambiguous. It also allows us in the future to build a single FSM for all language terminals.

Perhaps this bug was uncovered by #641 .. If I can suggest, how about we fix the `not_followed_by` parts of the existing/separate `TokenDefinition`s, so that they can stay mutually exclusive? for example:

```rust
not_followed_by = Choice([
    Fragment(IdentifierStart),
    Fragment(DecimalDigits),
    Fragment(Dot)
])
```